src/main/java/org/spongepowered/common/world/border/PlayerOwnBorderListener.java

```diff
@@ -0,0 +1,86 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.world.border;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.server.SPacketWorldBorder;
+import net.minecraft.world.border.IBorderListener;
+import net.minecraft.world.border.WorldBorder;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.common.mixin.core.entity.player.MixinEntityPlayerMP;
+
+public class PlayerOwnBorderListener implements IBorderListener {
+
+    private EntityPlayerMP player;
+
+    public PlayerOwnBorderListener(EntityPlayerMP player) {
+        this.player = player;
+    }
+
+    @Override
+    public void onSizeChanged(WorldBorder border, double newSize) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_SIZE));
+    }
+
+    @Override
+    public void onTransitionStarted(WorldBorder border, double oldSize, double newSize, long time) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.LERP_SIZE));
+    }
+
+    @Override
+    public void onCenterChanged(WorldBorder border, double x, double z) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_CENTER));
+    }
+
+    @Override
+    public void onWarningTimeChanged(WorldBorder border, int newTime) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_WARNING_TIME));
+    }
+
+    @Override
+    public void onWarningDistanceChanged(WorldBorder border, int newDistance) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_WARNING_BLOCKS));
+    }
+
+    @Override
+    public void onDamageAmountChanged(WorldBorder border, double newAmount) {
+    }
+
+    @Override
+    public void onDamageBufferChanged(WorldBorder border, double newSize) {
+    }
+    
+    /**
+     * This method is for cleaning up the player reference once they disconnect.
+     */
+    public void onPlayerDisconnect() {
+        ((net.minecraft.world.border.WorldBorder) ((MixinEntityPlayerMP) (Object) this.player).getWorldBorder().get()).listeners.remove(this);
```

Comment:
Have you tested this? You shouldn't be able to cast to a mixin class because they do not exist at runtime. Also you should make sure that the world border is actually still set to prevent errors (just replace `.get()` with `ifPresent` or with an additional `isPresent()` check)